src/main/java/com/matt/forgehax/mods/CoordsHud.java

```diff
@@ -11,85 +8,243 @@
-  protected double getDefaultScale() { return 1d; }
-  
+  protected double getDefaultScale() {
+    return 1d;
+  }
+
   double thisX;
   double thisY;
   double thisZ;
   double otherX;
   double otherZ;
-  
+
+  int posY;
+
   @SubscribeEvent
   public void onLocalPlayerUpdate(LocalPlayerUpdateEvent ev) {
-    if (getWorld() == null) return;
-  
-    EntityPlayerSP player = getLocalPlayer();
-    thisX = player.posX;
-    thisY = player.posY;
-    thisZ = player.posZ;
-    
-    double thisFactor = getWorld().provider.getMovementFactor();
+    if (MC.world == null) {
+      return;
+    }
+
+    switch (mode.get()) {
+      case VIEWENTITY: {
+        Entity viewEntity = getRenderEntity();
+        thisX = viewEntity.posX;
+        thisY = viewEntity.posY;
+        thisZ = viewEntity.posZ;
+        break;
+      }
+      case PLAYER: {
+        EntityPlayerSP player = getLocalPlayer();
+        thisX = player.posX;
+        thisY = player.posY;
+        thisZ = player.posZ;
+        break;
+      }
+    }
+
+    double thisFactor = MC.world.provider.getMovementFactor();
     double otherFactor = thisFactor != 1d ? 1d : 8d;
     double travelFactor = thisFactor / otherFactor;
     otherX = thisX * travelFactor;
     otherZ = thisZ * travelFactor;
   }
-  
+
   @SubscribeEvent
   public void onRenderOverlay(RenderGameOverlayEvent.Text event) {
     List<String> text = new ArrayList<>();
-    
-    if (!translate.get() || (translate.get() && multiline.get())) {
-      text.add(String.format("%01.1f, %01.0f, %01.1f", thisX, thisY, thisZ));
+	String facingNormal, facingWithTCoords, coordsNormal, coordsMultiTranslated, coordsTranslated;
+
+    if (brackets.get()) {
+      // Direction
+      facingNormal = String.format("%s " + "[%s]", getFacing(), getTowards());
+      // Multiline coords + direction
+      facingWithTCoords = String.format("[ %.1f ⏐ %.1f ] - %s [%s]", otherX, otherZ, getFacing(), getTowards());
+      // Only OW coords
+      coordsNormal = String.format("[ X %.1f ⏐ %.1f Z ] (%.0f Y)", thisX, thisZ, thisY);
+      // Multiline Nether coords
+      coordsMultiTranslated = String.format("[ %.1f ⏐ %.1f ]", otherX, otherZ);
+      // Single line OW + Nether coords
+      coordsTranslated = String.format("[ X %.1f ⏐ %.1f Z ] (%.0f Y)  %.1f ⏐ %.1f", thisX, thisZ, thisY, otherX, otherZ);
+    } else {
+      // Direction
+      facingNormal = String.format("%s " + "[%s]", getFacing(), getTowards());
+      // Multiline coords + direction
+      facingWithTCoords = String.format("%s " + "[%s] (%01.1f, %01.1f)", getFacing(), getTowards(), otherX, otherZ);
+      // Only OW coords
+      coordsNormal = String.format("%01.1f, %01.1f, %01.1f", thisX, thisY, thisZ);
+      // Multiline Nether coords
+      coordsMultiTranslated = String.format("(%01.1f, %01.1f)", otherX, otherZ);
+      // Single line OW + Nether coords
+      coordsTranslated = String.format("%01.1f, %01.1f, %01.1f (%01.1f, %01.1f)", thisX, thisY, thisZ, otherX, otherZ);
```

Comment:
I don't know about Java, but user defined format strings is one of the most known vulnerabilities in C, I don't think that would be a good idea. If the user put one extra %s, it would end up in an exception. Also it doesn't really provide more customization since you can't really print zero width or skip format arguments (afaik? please correct me if you can!). The "alternative layout" is unnecessary and can be removed, anyone can just put their own fancy fstrings and recompile, but the options to show various pieces aren't really replaceable with a user-defined fstring : I either put all the possible args, and user is stuck with them, or I put fewer, and user can't add more stuff.

Reply:
Kinda reverted to how it was before, implemented getPlayerDirection() from Helper class